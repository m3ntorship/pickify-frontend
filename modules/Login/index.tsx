export { default as Login } from './pages/Login';
