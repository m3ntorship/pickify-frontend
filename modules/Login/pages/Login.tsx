import React, { useEffect } from 'react';
import type { FC, ReactElement } from 'react';

import { loginUser } from '../../../context/AuthUserContext/api/authApi';
import { useRedirect } from '../../shared/hooks/useRedirect/useRedirect';
import { useAuth } from '../../../context/AuthUserContext/AuthUserContext';
import {
  getLastPage,
  clearLastPage,
} from '../../shared/logic/userAuth/userAuth';
import Logo from '../../shared/components/icons/logo.svg';
import Google from '../../shared/components/icons/google.svg';
import styles from './Login.module.css';
import Footer from '../../shared/components/molecules/Footer/Footer';
import Box from '../../shared/components/atoms/Box/Box';

const Login: FC = (): ReactElement => {
  const { loading, isAuthenticated } = useAuth();
  const { redirectToHomePage, redirectToPage } = useRedirect();

  useEffect(() => {
    if (!loading && isAuthenticated) {
      const lastPage = getLastPage();
      if (lastPage) {
        redirectToPage(lastPage);
        clearLastPage();
      } else {
        redirectToHomePage();
      }
    }
  }, [isAuthenticated, loading, redirectToHomePage, redirectToPage]);

  const login = async (): Promise<void> => {
    await loginUser();
  };
  return (
    <div className={styles['login-body-wrapper']}>
      <div className={styles['login-body']}>
        <div className={styles['main-wrapper']}>
          <header>
            <Logo />
          </header>
          <Box isWhiteColor>
            <Box.Body classes={styles.main}>
              <>
                <h1 className={styles['main-h1']}>Welcome to Pickify</h1>
                <p className={styles['main-paragraph']}>
                  Pickify is an online platform that helps people make better
                  decisions through voting insights
                </p>
                <button
                  type="button"
                  data-testid="login-test-button"
                  onClick={login}
                  className={styles['sign-in-button']}
                >
                  <div className={styles['google-logo']}>
                    <div>
                      <Google />
                    </div>
                  </div>
                  <span className={styles['sign-in-button-paragraph']}>
                    Sign in with Google
                  </span>
                </button>
              </>
            </Box.Body>
          </Box>
        </div>
        <div>
          <Footer showCenteredLogo />
        </div>
      </div>
    </div>
  );
};

export default Login;
