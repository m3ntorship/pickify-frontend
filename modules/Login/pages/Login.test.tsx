import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('Login page', () => {
  it('Snapshot testing for the login component', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
