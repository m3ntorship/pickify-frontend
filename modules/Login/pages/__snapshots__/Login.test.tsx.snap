// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Snapshot testing for the login component 1`] = `
<div
  className="login-body-wrapper"
>
  <div
    className="login-body"
  >
    <div
      className="main-wrapper"
    >
      <header>
        <svg
          data-file-name="SvgLogo"
        />
      </header>
      <div
        className="box bg-white"
      >
        <div
          className="main"
        >
          <h1
            className="main-h1"
          >
            Welcome to Pickify
          </h1>
          <p
            className="main-paragraph"
          >
            Pickify is an online platform that helps people make better decisions through voting insights
          </p>
          <button
            className="sign-in-button"
            data-testid="login-test-button"
            onClick={[Function]}
            type="button"
          >
            <div
              className="google-logo"
            >
              <div>
                <svg
                  data-file-name="SvgGoogle"
                />
              </div>
            </div>
            <span
              className="sign-in-button-paragraph"
            >
              Sign in with Google
            </span>
          </button>
        </div>
      </div>
    </div>
    <div>
      <footer
        className="footer"
      >
        <ul
          className="list-parent"
        >
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/credits"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Pickify Team
            </a>
          </li>
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Blog
            </a>
          </li>
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Careers
            </a>
          </li>
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Terms & Conditions
            </a>
          </li>
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Help
            </a>
          </li>
          <li
            className="link-item"
          >
            <a
              className="link"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Contact
            </a>
          </li>
        </ul>
        <div>
          <div
            className="text-center mt-4"
          >
            <h3
              className="copy-rights"
            >
              Pikcify Â© 2021
            </h3>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
